src/multistore.rs

```diff
@@ -0,0 +1,131 @@
+// mostly a copy of https://github.com/filecoin-project/merkle_light/blob/master/src/store/vec.rs
+
+use std::ops::Range;
+
+use anyhow::Result;
+
+use merkletree::merkle::Element;
+use merkletree::store::{DiskStore, Store, StoreConfig, VecStore};
+
+#[derive(Debug)]
+pub struct MultiStore<E: Element>(DiskStore<E>, VecStore<E>);
+
+const DISK_MAX: usize = 262144 * 48; // ~375Mb
+
+impl<E: Element> Store<E> for MultiStore<E> {
+    fn new_with_config(size: usize, _config: StoreConfig) -> Result<Self> {
+        Self::new(size)
+    }
+
+    fn new(size: usize) -> Result<Self> {
+        Ok(MultiStore(
+            DiskStore::new(DISK_MAX).unwrap(),
+            VecStore::new(size - DISK_MAX).unwrap(),
+        ))
+    }
+
+    fn write_at(&mut self, el: E, index: usize) -> Result<()> {
+        if index > DISK_MAX {
+            self.1.write_at(el, index - DISK_MAX)
+        } else {
+            self.0.write_at(el, index)
+        }
+    }
+
+    fn copy_from_slice(&mut self, buf: &[u8], start: usize) -> Result<()> {
+        if start + (buf.len() / E::byte_len()) > DISK_MAX {
+            self.1.copy_from_slice(buf, start - DISK_MAX)
+        } else {
+            self.0.copy_from_slice(buf, start)
+        }
+    }
+
+    fn new_from_slice_with_config(size: usize, data: &[u8], _config: StoreConfig) -> Result<Self> {
+        Self::new_from_slice(size, &data)
+    }
+
+    fn new_from_slice(_size: usize, _data: &[u8]) -> Result<Self> {
+        unimplemented!("nope, too hard");
+    }
+
+    fn new_from_disk(_size: usize, _config: &StoreConfig) -> Result<Self> {
+        unimplemented!("Cannot load a MultiStore from disk");
+    }
+
+    fn read_at(&self, index: usize) -> Result<E> {
+        if index > DISK_MAX {
+            self.1.read_at(index - DISK_MAX)
+        } else {
+            self.0.read_at(index)
+        }
+    }
+
+    fn read_into(&self, index: usize, buf: &mut [u8]) -> Result<()> {
+        if index > DISK_MAX {
+            self.1.read_into(index - DISK_MAX, buf)
+        } else {
+            self.0.read_into(index, buf)
+        }
+    }
+
+    fn read_range_into(&self, _start: usize, _end: usize, _buf: &mut [u8]) -> Result<()> {
+        unimplemented!("Not required here");
+    }
+
+    fn read_range(&self, r: Range<usize>) -> Result<Vec<E>> {
+        if r.start > DISK_MAX {
+            let nr = Range {
+                start: r.start - DISK_MAX,
+                end: r.end - DISK_MAX,
+            };
+            self.1.read_range(nr)
+        } else {
```

Comment:
This is just a personal taste thing. I would make it an `else if` as it is then clearly split into three parts:

 1. Fully in memory
 2. First few bytes on disks, rest on memory 
 3. All on disk